### ISSUE TITLE
Implement TextGrad (bounty)

### ISSUE THREAD

**Issue by abrichr** [OpenAdaptAI/OpenAdapt#748]
### Feature request

We would like to implement https://github.com/zou-group/textgrad.

### Motivation

> It works out-of-the-box for a variety of tasks,
where the users only provide the objective function without tuning components or prompts of
the framework.

<img width="125" alt="image" src="https://github.com/OpenAdaptAI/OpenAdapt/assets/774615/45811761-a873-4f80-9cc4-6fa07e52829d">

### Bounty

A paid bounty is available. Please suggest a price range 🙏

**Comment by kernel-loophole** [comment]
@abrichr wanted to work on this in $750.looking forward to discuss more about it .

**Comment by abrichr** [comment]
Sounds good @kernel-loophole ! Please DM me on Discord 🙏

**Comment by kernel-loophole** [comment]
@abrichr can you share your discord username

**Comment by kernel-loophole** [comment]
@abrichr send you request on discord

**Comment by kernel-loophole** [comment]
@abrichr  I will implement TextGrad into OpenAdapt, leveraging its framework for textual gradients to optimize and automate GUI workflows with Large Multimodal Models (LMMs). By integrating TextGrad's natural-language loss function and optimization through textual feedback, I aim to enhance process automation in OpenAdapt, allowing AI models to iteratively refine solutions based on GUI inputs, similar to backpropagation in neural networks. This will improve task accuracy, reduce repetitive manual input, and optimize performance across various desktop and web-based automation tasks.

**Comment by abrichr** [comment]
Hi @kernel-loophole , thank you for your interest. Can you please be more precise? e.g. where will you implement textgrad? How will you evaluate whether the implementation has been successful?

**Comment by kernel-loophole** [comment]
@abrichr currently i do know the repo code architecture but i think where you pass the LLM the input ,the textgrad will be implement in that code where model can learn from backpropagation .and for evaluation we can use the same llm with textgrad and without textgrad ,then compares the performance

**Comment by omarsoufiane** [comment]
hello @abrichr , you can create a bounty on [Bountyhub](https://www.bountyhub.dev/), it's integrated into github, (a bot will comment to let other people know that there is a bounty) and if no one solves your bounty you can retract it and get your money back, it's a simple process where you import issues by pasting the issue URL and the payment is upfront so that developers attempting to solve the issue know that the amount is already available.

**Comment by kernel-loophole** [comment]
hi there @abrichr have not hear from you on this, any suggestions or feedback where i can start .

**Comment by abrichr** [comment]
Closing as part of the architecture transition to modular packages. The legacy codebase has been frozen at v0.46.0.

TextGrad implementation would be considered in the new openadapt-ml package.

For the new modular architecture, please open issues in the relevant package repository:
- GUI capture: https://github.com/OpenAdaptAI/openadapt-capture
- ML/training: https://github.com/OpenAdaptAI/openadapt-ml
- Benchmarks: https://github.com/OpenAdaptAI/openadapt-evals
- Privacy: https://github.com/OpenAdaptAI/openadapt-privacy

See PR #960 for details on the new architecture.
